snuba/query/parser/expressions.py

```diff
@@ -1,28 +1,78 @@
-from typing import Any, Optional
+import re
 
-from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from typing import Any, Iterable, Optional
+
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
 from snuba.query.parser.functions import parse_function_to_expr
 from snuba.util import is_function, QUOTED_LITERAL_RE
 
+# We still have some cases were people pass clickhouse expressions in the
+# aggregate clause of the Snuba query since snuba does not reject them.
+# In order to process the aggregate first we need to turn those expressions
+# into Snuba functions.
+# This regex provides some very basic support to decompose those expressions.
+AGGR_CLICKHOUSE_FUNCTION_RE = re.compile(
+    r"^(?P<name>[a-zA-Z_][a-zA-Z0-9_]*)\s*\((?P<params>[\w\,\.\s]*)\)$"
+)
+
 
 def parse_expression(val: Any) -> Expression:
     if is_function(val, 0):
         return parse_function_to_expr(val)
     # TODO: This will use the schema of the dataset to decide
     # if the expression is a column or a literal.
-    if QUOTED_LITERAL_RE.match(val):
-        return Literal(None, val[1:-1])
+    if val.isdigit():
+        return Literal(None, int(val))
     else:
-        return Column(None, val, None)
+        try:
+            return Literal(None, float(val))
+        except Exception:
+            if QUOTED_LITERAL_RE.match(val):
+                return Literal(None, val[1:-1])
+            else:
+                return Column(None, val, None)
```

Comment:
Does this mean that any value that isn't a number or passed with single quotes gets treated as a column?

Reply:
Yes. The main issues are that, on one side the snuba syntax itself does not allow us to distinguish between a column, a table, or another clickhouse keyword/identifier, on the other side this is not meant to be a comprehensive clickhouse parser. We should not allow users to provide freeform expressions. So this is the minimum to be backwards compatible with what we receive now without encouraging more freeform expressions.